src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -254,7 +258,22 @@ impl<'tcx> RegionInferenceContext<'tcx> {
                 debug!("classify_constraint: terminator.kind={:?}", terminator.kind);
                 match terminator.kind {
                     TerminatorKind::DropAndReplace { .. } => ConstraintCategory::Assignment,
-                    TerminatorKind::Call { .. } => ConstraintCategory::CallArgument,
+                    TerminatorKind::Call { destination: Some((ref place, _)), .. } => {
+                        if tcx.any_free_region_meets(
```

Comment:
I'd like a comment here... something lke:

---

Classify calls differently depending on whether or not the `sub` region appears in the return type. If the return type contains the sub-region, then this is either an assignment or a return, depending on whether we are writing to the `RETURN_PLACE` or not. The idea here is that the region is being propagated from an input into the output place, so it's a kind of assignment. Otherwise, if the sub-region only appears in the argument types, then use the `CallArgument` classification.